IPs/RSKIP186.md

```diff
@@ -0,0 +1,70 @@
+# Active Federation creation block height registration
+
+|RSKIP          |186           |
+| :------------ |:-------------|
+|**Title**      |Active Federation creation block height registration |
+|**Created**    |19-NOV-20 |
+|**Author**     |JD |
+|**Purpose**    |Sec |
+|**Layer**      |Core |
+|**Complexity** |1 |
+|**Status**     |Draft |
+
+## Abstract
+
+When the Federation change voting process finishes, the Bridge emmits a `commit_federation` event.
+This event can be used to provide a blockchain proof of the Federation address when requested. Nowadays doing this would imply navigating the blockchain to determine when it was emmited, or constantly monitoring said events.
+
+This RSKIP proposes adding on chain information for easily fetch the block where the Federation was changed for the last time. With the block, a system could provide the proof that it belongs to the mainchain.
+
+## Motivation
+
+When a user gets the Federation address they usually interact with a node/service and have to trust on this information. This exposes a risk where a man-in-the-middle kind of attack could lead users to send their funds to invalid address thus losing them.
+
+## Specification
+
+### Storage changes
+
+The Bridge will store two new values:
+- activeFederationCreationBlockHeight. (long)
+  - Defaults to the existing Federation activation height 
+- nextFederationCreationBlockHeight. (long)
+  - Defaults to the existing Federation activation height
```

Comment:
The names sound a bit confusing to me, I had to read a couple of times what the difference between each is.

Maybe:
```
federationCreationBlockHeight
federationActivationBlockHeight
```

What do you think?

Reply:
- activeFederationCreationBlockHeight: stores the active Federation creation height. This value is the one the user would get
- nextFederationCreationBlockHeight: this is a temporary value created when a new Federation is committed. This committed Federation won't be activated until the network's configuration delay is achieved.

I'll try better explaining this in the RSKIP.

Thanks!